jax/_src/lax/control_flow/for_loop.py

```diff
@@ -70,21 +70,23 @@ class Ref(Generic[T]): pass
 ### Tracing utilities
 
 def _hoist_consts_to_refs(jaxpr: core.Jaxpr) -> core.Jaxpr:
-  num_consts = len(jaxpr.constvars)
+  all_const_avals = [var.aval for var in jaxpr.constvars]
```

Comment:
Note to future selves: this function used to take a jaxpr with refs and non-ref inputs, where the non-refs were exactly closed-over consts, and produce a jaxpr with only ref inputs where the consts were read. But after this change, closed-over things could be refs (not just non-refs) introduced by an outer `for` loop, so the type signature is the same but in the implementation we don't want to e.g. wrap refs in refs.